server/src/main/java/org/opensearch/identity/IdentityModule.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.identity;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchException;
+import org.opensearch.identity.noop.NoopIdentityPlugin;
+import java.util.List;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.plugins.IdentityPlugin;
+import java.util.stream.Collectors;
+
+/**
+ * Identity and access control for OpenSearch.
+ *
+ * @opensearch.experimental
+ * */
+public class IdentityModule {
+    private static final Logger logger = LogManager.getLogger(IdentityModule.class);
+
+    /** Application wide access for identity systems */
+    private static IdentityModule IDENTITY_MODULE = null;
+
+    /**
+     * Gets the Identity Service for this application
+     */
+    public static IdentityModule getModule() {
```

Comment:
Looks like this is going for the singleton pattern? What do you think about creating an `init` function that initializes the parameters and follows the singleton convention of `getInstance()` (https://stackoverflow.com/a/1051076)

```
private static IdentityModule INSTANCE;

private IdentityModule() {}

public static IdentityModule getInstance() {
    if (INSTANCE == null) {
        INSTANCE = new IdentityModule();
    }

    return INSTANCE;
}

public static void init(final Settings settings, final List<IdentityPlugin> identityPlugins) { ... }
```

A singleton class with a public constructor is not a singleton.

Reply:
Good callout, I checked and I didn't really need the static accessor.  Shifting to pass the instance via constructor/guice injection.